test/Mono.Linker.Tests.Cases/DataFlow/CompilerGeneratedStateMachine.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	[ExpectedNoWarnings]
+	[SkipKeptItemsValidation]
+	class CompilerGeneratedStateMachine
+	{
+		public static void Main ()
```

Comment:
Nit: I would add couple more negative tests (which produce diagnostics) - especially in analyzers it's important since we give up on analysis if we don't understand something and won't produce any diagnostics. So the fact that a success test passed doesn't necessarily mean the analyzer worked.
Linker is the other way round, so success tests are much more "reliable".